geotrellis/geotrellis-server
# [WIP] Attempt to abstract Repository over F

## Overview

Here's progress on this task up to this point. As of 06/05/2020 this task is paused to focus on other high priority work.

Currently does not fully compile, there are two major remaining tasks to address:
- [ ] Implement the stubs for the MapAlgebraStacOgcRepositor[y|ies] `find()` methods 
- [ ] Address is the inclusion of `F` in the `Repository` argument currently passed to `WcsModel`, `WmsModel`, and `WmtsModel`

In addition we need to test sufficiently enough to verify that we haven't broken the name shuffling logic in the MapAlgebraStacRepository implementations.

### Checklist


### Demo

Optional. Screenshots, `http` examples, etc.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

- How to test this PR
- Prefer bulleted description
- Start after checking out this branch
- Include any setup required, such as bundling scripts, restarting services, etc.
- Include test case, and expected output

Closes #azavea/geotrellis#268
